GeneralsMD/Code/Tools/wdump/chunk_d.cpp

```diff
@@ -1183,6 +1247,24 @@ void	ChunkTableClass::List_W3D_CHUNK_SCG(ChunkItem * Item,CListCtrl *List)
 		data++;
 	}
 }
+
+void	ChunkTableClass::List_W3D_CHUNK_FXSHADER_IDS(ChunkItem* Item, CListCtrl* List)
+{
+	int Counter = 0;
+	int counter = 0;
+	uint32* data = (uint32*)Item->Data;
+	void* max = (char*)Item->Data + Item->Length;
+	char buf[256];
+
+	while (data < max) {
+
+		sprintf(buf, "Vertex[%d] FXShader Index", counter);
```

Comment:
Prefer use `snprintf` which takes buffer size too.

Reply:
Ok in that case it is fine. We can make a pass in a future change and replace unsafe C functions.